packages/apps/mercadopago/src/mp-create-transaction.ts

```diff
@@ -0,0 +1,291 @@
+import type { AppModuleBody } from '@cloudcommerce/types';
+import type{ CreateTransactionParams } from '@cloudcommerce/types/modules/create_transaction:params';
+import type{ CreateTransactionResponse } from '@cloudcommerce/types/modules/create_transaction:response';
+import type { Firestore } from 'firebase-admin/firestore';
+import { logger } from 'firebase-functions';
+import config from '@cloudcommerce/firebase/lib/config';
+import axios from 'axios';
+
+const locationId = config.get().httpsFunctionOptions.region;
+const baseUri = `https://${locationId}-${process.env.GCLOUD_PROJECT}.cloudfunctions.net`;
```

Comment:
Se você executa `config.get()` fora de alguma função ele não vai ser reativo a `config.set`.
Quer dizer, se aqui https://github.com/ecomplus/store/blob/40e5cf081aa4e8f26129dfda0a2ddc1b1f5eeb8b/functions/events/index.js#L4-L7 a configuração alterasse a região, você não pegaria a região correta, pegaria sempre a default.
O módulo vai ser importado e cacheado antes desse trecho `config.set`, mas a função não vai ser executada antes...

Reply:
Isso também vale pra outras props de configuração "global", e também pra `getEnv`...
"Sempre" dentro de alguma função:

```ts
// x.ts

export const x = () => {
  const { storeId } = config.get();
};
```

"Nunca" no escopo do módulo:

```ts
// x.ts

const { storeId } = config.get();

export const x = () => {
  console.log(`Este Store ID vai estar errado ${storeId} :P`):
};
```

A única exceção é quando a config é necessária pra algum export, nesse caso tem que tomar cuidado quando for importar o módulo depois, `import/first` pode vai ter que ser gambelado

Reply:
> Se lembrar de outro lugar onde fez isso ... 🙃